app/services/GZipService.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import java.io.ByteArrayInputStream
+import java.util.zip.GZIPInputStream
+
+import models.ResponseErrorMessage
+import models.ResponseErrorType.OtherError
+import play.api.libs.json.JsValue
+import play.api.libs.json.Json
+
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+object GZipService {
+
+  def decompressArrayByteToJson(arrayByte: Array[Byte]): Either[ResponseErrorMessage, JsValue] =
+    Try {
+      val inputStream = new GZIPInputStream(new ByteArrayInputStream(arrayByte))
+      Json.parse(scala.io.Source.fromInputStream(inputStream).mkString)
+    } match {
+      case Success(value) => Right(value)
+      case Failure(_)     => Left(ResponseErrorMessage(OtherError, None))
+    }
```

Comment:
Really clean and easy. I also like that you're thinking about error handling here, looks good 👍